lib/livebook/notebook/explore/intro_to_nx.livemd

```diff
@@ -1,3 +1,732 @@
+Add your function directly to the previous Elixir cell.
+
+<!-- livebook:{"break_markdown":true} -->
+
+### Using tensor aware operators
+
+<!-- livebook:{"break_markdown":true} -->
+
+That bit of code does nothing but wrap the `-` operator. It's tensor aware. 
+We can use it to do several different mathematical calculations. The most
+obvious is to subtract two matrices with like shapes, like this:
+
+```elixir
+TensorMath.subtract(tensor2, tensor)
+```
+
+In this program, `Nx.Defn.Kernel.-/2` replaces `Kernel.-/2`, so
+we predictably get the value of `Nx.Defn.Kernel.-(tensor2, tensor)`.
+You can also process this equation:
+
+<!-- livebook:{"break_markdown":true} -->
+
+$$
+\begin{bmatrix}
+  1 & 2 \\
+  3 & 4
+\end{bmatrix}
+
+-
+1
+
+\begin{bmatrix}
+  0 & 1 \\
+  2 & 3
+\end{bmatrix}
+$$
+
+<!-- livebook:{"break_markdown":true} -->
+
+Mathematically, it's the same as this:
+
+$$
+\begin{bmatrix}
+  1 & 2 \\
+  3 & 4
+\end{bmatrix}
+
+-
+\begin{bmatrix}
+  1 & 1 \\
+  1 & 1
+\end{bmatrix}
+
+\begin{bmatrix}
+  0 & 1 \\
+  2 & 3
+\end{bmatrix}
+$$
+
+<!-- livebook:{"break_markdown":true} -->
+
+`defn` takes care of this rewrite implicitly
+through a broadcast. Nx will do this under the
+hood, like this:
+
+```elixir
+Nx.broadcast(1, {2, 2})
+```
+
+This broadcast takes the scalar `1` and translates it
+to a compatible shape by copying it. Of course, this
+all happens automatically:
+
+```elixir
+TensorMath.subtract(tensor, 1)
+```
+
+Or, you could subtract a tensor from a scalar:
+
+```elixir
+TensorMath.subtract(10, tensor)
+```
+
+Or subtract a row or column. Mathematically, it would look like this:
+
+<!-- livebook:{"break_markdown":true} -->
+
+$$
+\begin{bmatrix}
+  1 & 2 \\
+  3 & 4
+\end{bmatrix}
+
+-
+\begin{bmatrix}
+  1 & 2
+\end{bmatrix}
+
+\begin{bmatrix}
+  0 & 0 \\
+  2 & 2
+\end{bmatrix}
+$$
+
+<!-- livebook:{"break_markdown":true} -->
+
+which is the same as this:
+
+<!-- livebook:{"break_markdown":true} -->
+
+$$
+\begin{bmatrix}
+  1 & 2 \\
+  3 & 4
+\end{bmatrix}
+
+-
+\begin{bmatrix}
+  1 & 2 \\
+  1 & 2
+\end{bmatrix}
+
+\begin{bmatrix}
+  0 & 0 \\
+  2 & 2
+\end{bmatrix}
+$$
+
+<!-- livebook:{"break_markdown":true} -->
+
+This rewrite happens in Nx too, also through a broadcast. We want to
+broadcast the tensor `[1, 2]` to match the `{2, 2}` shape, like this:
+
+```elixir
+Nx.broadcast(Nx.tensor([1, 2]), {2, 2})
+```
+
+The `-` operator in `Nx` takes care of that broadcast
+implicitly, as before:
+
+```elixir
+TensorMath.subtract(tensor, Nx.tensor([1, 2]))
+```
+
+In the notebooks to come, we'll walk through the `Nx.broadcast/2`
+function that resolves two distinct shapes into compatible tensors.
+For now, know that Nx uses `broadcast` to repeat like rows or scalars
+until shapes match. Read more about it here:
+
+```elixir
+h(Nx.broadcast())
+```
+
+You now know how to build custom tensor-aware code.
+The next step is to make the code efficient on specialized CPUs
+using the backend of your choice.
```

Comment:
I love the explanation of broadcasts, especially with the latex graphs! :+1: I think broadcast definitely deserves its own section. What if we talk about Nx.add/Nx.subtract first, introduce the concepts of broadcast, and then jump into `defn`?